modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -961,11 +978,35 @@ private HybridTimestamp createBeginTimestampWithIncrementRwTxCounter() {
         });
     }
 
+    /** Called when a read-write transaction is finished. */
+    private void onFinishRwTx(UUID txId) {
+        decrementRwTxCount(txId);
+
+        unregister(txId);
+    }
+
     private void decrementRwTxCount(UUID txId) {
         localRwTxCounter.inUpdateRwTxCountLock(() -> {
             localRwTxCounter.decrementRwTxCount(beginTimestamp(txId));
 
             return null;
         });
     }
+
+    /**
+     * Puts a transaction into the registry.
+     *
+     * @param tx Transaction.
+     * @return Registered transaction.
+     */
+    private InternalTransaction register(InternalTransaction tx) {
+        transactions.put(tx.id(), tx);
```

Comment:
let's add an assertion to make sure we are not overriding another transaction here (e.g. tx is registered only once; there is no duplicate ids at least locally)

Reply:
The PR was reworked so that the system view now uses existing collections. Review it again, please.